linkerd/proxy/http/src/classify/insert.rs

```diff
@@ -61,8 +67,9 @@ where
 
     fn proxy(&self, svc: &mut S, mut req: http::Request<B>) -> Self::Future {
         let classify_rsp = self.classify.classify(&req);
-        let prior = req.extensions_mut().insert(classify_rsp);
-        debug_assert!(prior.is_none(), "classification extension already existed");
+        if req.extensions_mut().insert(classify_rsp).is_some() {
+            tracing::debug!("Overrode response classifier");
```

Comment:
nit, take it or leave it: maybe worth having this print what the previous classifier was?
```suggestion
        if let Some(prev) = req.extensions_mut().insert(classify_rsp) {
            tracing::debug!(?prev, "Overrode response classifier");
```

Reply:
that requires adding `Debug` bounds which felt a little larger than desired.